torch/csrc/jit/ir/ir.cpp

```diff
@@ -1595,6 +1595,18 @@ Node* Graph::createTupleSlice(Value* tup, int64_t beg, int64_t end) {
   return n;
 }
 
+Node* Graph::createEnumName(Value* e) {
+  auto n = create(prim::EnumName, {e});
+  n->output()->setType(StringType::get());
+  return n;
+}
+
+Node* Graph::createEnumValue(Value* e, const TypePtr& enum_val_type) {
```

Comment:
Consider adding an assert here to check that `e`'s type is an enum and that `enum_val_type` is an enum type. Actually, I think you can pass in only `e` and derive `enum_val_type` from it.

Reply:
Done.

Reply:
Why did you choose to pass in `enum_val_type` explicitly instead of deriving it from `e`? Am I looking at an old version?

Reply:
Mostly to follow convention and to save one 'cast'. But I guess the second reason is too minute and voided by having an assert there. Done.